contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -134,23 +134,47 @@ def resolve_target(self, node_task, target, results_dir, node_paths, resolve_loc
           raise TaskError('Failed to resolve dependencies for {}:\n\t{} failed with exit code {}'
                           .format(target.address.reference(), command, result))
         if source_deps:
-          self._link_source_dependencies(target, results_dir, node_paths, source_deps)
-
-  def _link_source_dependencies(self, target, results_dir, node_paths, source_deps):
+          self._link_source_dependencies(node_task, target, results_dir, node_paths, source_deps)
 
+  def _link_source_dependencies(self, node_task, target, results_dir, node_paths, source_deps):
     for package_name, file_path in source_deps.items():
       # Package name should always the same as the target name
       dep = self._get_target_from_package_name(target, package_name, file_path)
+
+      # Apply node-scoping rules if applicable
+      node_scope = dep.payload.node_scope or node_task.node_distribution.node_scope
```

Comment:
I'm a little surprised that the node_distribution is pulled from the node task instead of using `global_instance()` on `NodeDistribution` for getting the `node_scope`.

It looks like `NodeDistribution` can be task scoped. It's a little confusing to me because I think that means that you could have multiple values of the default `node_scope` under different option scopes.

Another option would be to move this or'd clause to an `@property` on the node target class and add a subsystem dependency on `NodeDistribution` to the target.

Reply:
> It looks like NodeDistribution can be task scoped. 

Any subsystem can be scoped or unscoped... how it is consumed is convention. But you're right that this should probably be global.

That would be a separate change/issue though, and shouldn't happen here probably.

Reply:
I'll need to look into why we scope NodeDistribution at the task-level, but it does seem more reasonable to have it at the global level. I believe this may stem from some old history to support multiple versions of Node.js within the same repo.

Reply:
Hm. Ok.

Reply:
Yes please.